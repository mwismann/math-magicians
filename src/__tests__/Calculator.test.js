import { render, fireEvent, screen, } from "@testing-library/react";
import CalculatorPage from "../components/Calculator";

test('Calculator renders correctly', () => {
  const text = render(<CalculatorPage />);
  expect(text.toJSON).toMatchSnapshot();
});

test('Test calculator UI', async () => {
  render(<CalculatorPage />);
  fireEvent.click(screen.getByText('5'));
  let testDisplay = screen.getByTestId('display').textContent;
  expect(testDisplay).toMatch('5');
  fireEvent.click(screen.getByText('6'));
  testDisplay = screen.getByTestId('display').textContent;
  expect(testDisplay).toMatch('56');
  fireEvent.click(screen.getByText('+'));
  testDisplay = screen.getByTestId('display').textContent;
  expect(testDisplay).toMatch('56+');
  fireEvent.click(screen.getByText('2'));
  testDisplay = screen.getByTestId('display').textContent;
  expect(testDisplay).toMatch('56+2');
  fireEvent.click(screen.getByText('='));
  testDisplay = screen.getByTestId('display').textContent;
  expect(testDisplay).toMatch('58');
  fireEvent.click(screen.getByText('AC'));
  testDisplay = screen.getByTestId('display').textContent;
  expect(testDisplay).toMatch('');
});
