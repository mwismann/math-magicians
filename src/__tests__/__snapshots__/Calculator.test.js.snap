// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `undefined`;
